nuxeo-services/nuxeo-permissions/src/main/java/org/nuxeo/ecm/permissions/TransientUserPermissionHelper.java

```diff
@@ -19,34 +19,95 @@
 
 package org.nuxeo.ecm.permissions;
 
+import org.nuxeo.ecm.core.api.CoreInstance;
 import org.nuxeo.ecm.core.api.DocumentModel;
 import org.nuxeo.ecm.core.api.NuxeoPrincipal;
 import org.nuxeo.ecm.core.api.security.ACE;
 import org.nuxeo.ecm.core.api.security.ACL;
 import org.nuxeo.ecm.core.api.security.ACP;
+import org.nuxeo.ecm.core.query.sql.NXQL;
 import org.nuxeo.ecm.tokenauth.service.TokenAuthenticationService;
 import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.services.config.ConfigurationService;
 
 /**
  * @since 8.1
  */
 public class TransientUserPermissionHelper {
 
+    /**
+     * @since 10.3
+     */
+    public static final String OTHER_DOCUMENT_WITH_PERMISSION_QUERY = "SELECT ecm:uuid FROM Document"
+            + " WHERE (ecm:acl/*1/status is NULL OR ecm:acl/*1/status = 0 OR ecm:acl/*1/status = 1)"
```

Comment:
Perhaps add a comment explaining that status = 0 is PENDING and 1 is EFFECTIVE and that the goal of this is to exclude ARCHIVED ACLs